autotest/pymod/gdaltest.py

```diff
@@ -1938,6 +1938,45 @@ def tempfile(filename, content):
         gdal.Unlink(filename)
 
 
+###############################################################################
+# Temporarily set an attribute filter
+
+
+@contextlib.contextmanager
+def attribute_filter(lyr, filter_txt):
+    lyr.SetAttributeFilter(filter_txt)
+    try:
+        yield
+    finally:
+        lyr.SetAttributeFilter(None)
+
+
+###############################################################################
+# Temporarily set a spatial filter
+# Single argument is parsed as WKT or assumed to be an OGRGeometry
+# Four arguments are interpreted as bounding rectangle
+
+
+@contextlib.contextmanager
+def spatial_filter(lyr, *args):
```

Comment:
could we move that to the ogrtest module instead ?

```suggestion
###############################################################################
# Temporarily set an attribute filter


@contextlib.contextmanager
def attribute_filter(lyr, filter_txt):
    lyr.SetAttributeFilter(filter_txt)
    try:
        yield
    finally:
        lyr.SetAttributeFilter(None)


###############################################################################
# Temporarily set a spatial filter
# Single argument is parsed as WKT or assumed to be an OGRGeometry
# Four arguments are interpreted as bounding rectangle


@contextlib.contextmanager
def spatial_filter(lyr, *args):
```

Reply:
Yes, that seems like the more natural home. Though I had wondered whether the separation between `gdaltest` and `ogrtest` is useful at this point or if `ogrtest` could just be folded into `gdaltest`.

Reply:
having both modules has the advantage of having 2 smaller files rather than a larger one

Reply:
I've moved it over.